pkg/interceptors/interceptors.go

```diff
@@ -159,7 +162,31 @@ func ResolveToURL(getter InterceptorGetter, name string) (*apis.URL, error) {
 	return ic.ResolveAddress()
 }
 
-func Execute(ctx context.Context, client *http.Client, req *triggersv1beta1.InterceptorRequest, url string) (*triggersv1beta1.InterceptorResponse, error) {
+func Execute(ctx context.Context, client *http.Client, req *triggersv1beta1.InterceptorRequest, url string,
+	getter InterceptorGetter, name string) (*triggersv1beta1.InterceptorResponse, error) {
+
+	ic, err := getter(name)
+	if err != nil {
+		return nil, fmt.Errorf("url resolution failed for interceptor %s with: %w", name, err)
+	}
+	// There is a chances that url may come from ClusterInterceptorSpec/ClientConfig of URL or Service
+	// So adding check to make sure its https before creating client for https.
+	if strings.Contains(url, "https") {
+		certPool := x509.NewCertPool()
+		if ok := certPool.AppendCertsFromPEM(ic.Spec.ClientConfig.CaBundle); !ok {
+			return nil, fmt.Errorf("unable to parse cert from %s", ic.Spec.ClientConfig.CaBundle)
+		}
+
+		client = &http.Client{
```

Comment:
I think we have to update this - creating a new client here means we will ignore all of the config options that we set in the already passed in HTTP client (such as the timeout values). Creating a new HTTP client per call means we can't reuse the underlying connections so we might be limiting the number of calls from the EL to the interceptor.

I think we'd have to add in the certs in the `sinker.Start` function - we could pass in a `clusterinterceptorlister` as an arg to the sinker struct - list all the clusterinterceptors, get the caCert and add it to the certPool. and then create a httpClient with all the certs. (This does mean that if we add a new interceptor in the middle, we'd have to restart each EL.


(Ideally in the future we'd have something that listens for changes to interceptors and updates the httpClient with newCerts on the fly)